connect/api/src/main/java/org/apache/kafka/connect/connector/Connector.java

```diff
@@ -121,4 +125,23 @@ public void reconfigure(Map<String, String> props) {
      * Stop this connector.
      */
     public abstract void stop();
+
+    /**
+     * Validate the connector configuration values against configuration definitions.
+     * @param connectorConfigs the provided configuration values
+     * @return List of Config, each Config contains the updated configuration information given
+     * the current configuration values.
+     */
+    public Config validate(Map<String, String> connectorConfigs) {
+        ConfigDef configDef = defineConfig();
+        List<String> groups = configDef.groups();
+        List<ConfigValue> configValues =  configDef.validate(connectorConfigs);
+        return new Config(configDef, groups, configValues);
+    }
+
+    /**
+     * Define the configuration for the connector.
+     * @return The ConfigDef for this connector.
+     */
+    protected abstract ConfigDef defineConfig();
```

Comment:
Can we name this more like an accessor for Kafka, e.g. just `config()` instead of `defineConfig()`?